app/controllers/responses_controller.rb

```diff
@@ -0,0 +1,21 @@
+class ResponsesController < ApplicationController
+  def create
+    @question = Question.find(params[:question_id])
+    @response = Response.new(response_params)
+    @response.question = @question
+    @response.user = current_user if current_user
+    if @response.save
+      flash[:notice] = 'Response successfully added'
+      redirect_to @question
+    else
+      flash[:error] = @response.errors.full_messages.join(' ')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.